dynawo/sources/Modeler/DataInterface/DYNGeneratorInterfaceIIDM.cpp

```diff
@@ -170,10 +170,18 @@ GeneratorInterfaceIIDM::getQMax() {
     return generatorIIDM_.minMaxReactiveLimits().max();
   } else if (generatorIIDM_.has_reactiveCapabilityCurve()) {
     double qMax = 0;
-    for (unsigned int i = 0; i <= generatorIIDM_.reactiveCapabilityCurve().size(); ++i) {
-      IIDM::ReactiveCapabilityCurve::point const& current_point = generatorIIDM_.reactiveCapabilityCurve()[i];
-      if (current_point.qmax > qMax)
-        qMax = current_point.qmax;
+    if (- getP() < generatorIIDM_.reactiveCapabilityCurve()[0].p) {
+      qMax = generatorIIDM_.reactiveCapabilityCurve()[0].qmax;
+    } else if (- getP() > generatorIIDM_.reactiveCapabilityCurve()[generatorIIDM_.reactiveCapabilityCurve().size() - 1].p) {
+      qMax = generatorIIDM_.reactiveCapabilityCurve()[generatorIIDM_.reactiveCapabilityCurve().size() - 1].qmax;
+    } else {
+      for (unsigned int i = 0; i <= generatorIIDM_.reactiveCapabilityCurve().size() - 2; ++i) {
+        IIDM::ReactiveCapabilityCurve::point current_point = generatorIIDM_.reactiveCapabilityCurve()[i];
```

Comment:
const IIDM::ReactiveCapabilityCurve::point& (same below)

Reply:
I have tried this, and the values that I get are modified. It seems that I cannot access current_point.p and current_point_qmin with this. And I don't understand why.

Reply:
?! on this case let's at least try to store generatorIIDM_.reactiveCapabilityCurve() into a reference

Reply:
Yes, it is quite odd, I get different QMax and QMin. In fact when I call current_point.p and current_point.qmin, I obtain 0 which is not what is expected (it works fine with next_point.p, next_point.qmin, next_point.qmax and current_point.qmax).

Reply:
To store generatorIIDM_.reactiveCapabilityCurve() into a reference, I need to add a const IIDM::ReactiveCapabilityCurve& Name_of_ref = generatorIIDM_.reactiveCapabilityCurve() at the beginning and then use Name_of_ref instead ?

Reply:
yes exactly!